### ISSUE TITLE
Fetching from Pact Broker does not work in 4.1.8 version

### ISSUE THREAD

**Issue by anddreiu** [DiUS/pact-jvm#1227]
Hi,

I've just seen that there is an issue when fetching the pact files from Pact Broker using 4.1.8 version. I am using Junit runner and I get this error now:

~~~
2020-10-15 07:43:28.567 DEBUG [PactBrokerLoader.kt] Loading pacts from pact broker for provider providerName and consumer version selectors [ConsumerVersionSelector(tag=null, latest=true, consumer=consumerName)]
2020-10-15 07:43:28.573 DEBUG [PactBrokerLoader.kt] Authentication: Basic
2020-10-15 07:43:28.579 DEBUG [HalClient.kt] Fetching: /
kotlin.KotlinNullPointerException
	at au.com.dius.pact.core.pactbroker.PactBrokerClient$fetchConsumersWithSelectors$1.invoke(PactBrokerClient.kt:180)
	at au.com.dius.pact.core.pactbroker.PactBrokerClient$fetchConsumersWithSelectors$1.invoke(PactBrokerClient.kt:103)
	at au.com.dius.pact.core.support.KotlinLanguageSupportKt.handleWith(KotlinLanguageSupport.kt:36)
	at au.com.dius.pact.core.pactbroker.PactBrokerClient.fetchConsumersWithSelectors(PactBrokerClient.kt:176)
	at au.com.dius.pact.provider.junitsupport.loader.PactBrokerLoader.loadPactsForProvider(PactBrokerLoader.kt:194)
	at au.com.dius.pact.provider.junitsupport.loader.PactBrokerLoader.load(PactBrokerLoader.kt:111)
	at au.com.dius.pact.provider.junit.PactRunner.initialize(PactRunner.kt:88)
	at au.com.dius.pact.provider.junit.PactRunner.getChildren(PactRunner.kt:143)
	at org.junit.runners.ParentRunner.getFilteredChildren(ParentRunner.java:426)
	at org.junit.runners.ParentRunner.getDescription(ParentRunner.java:351)
~~~

I am filtering by provider and consumer names in my class. Could this be related to the change for this issue: https://github.com/DiUS/pact-jvm/issues/1193 ?

Thanks,
Andrei

**Comment by anto-ac** [comment]
@anddreiu It must be related, yes. @uglyog would know best on this.

What version of the broker are you using by the way?

**Comment by anddreiu** [comment]
Pact Broker version: 2.59.2

**Comment by anto-ac** [comment]
@anddreiu Would you be able to try with the latest version of the broker, even if it's not the proper solution.

**Comment by anddreiu** [comment]
Unfortunately I am not able to try this, as the Pact Broker service is owned by a different team, they will need to plan this and so on.

**Comment by anto-ac** [comment]
@anddreiu I am afraid you might have to downgrade your version of pact-jvm until a fix is out.

**Comment by anddreiu** [comment]
yep, not a problem. There is also a previous issue that is blocking me to upgrade the pact-jvm version to the latest: https://github.com/DiUS/pact-jvm/issues/1218

**Comment by bethesque** [comment]
@anddreiu can you provide some more information to help identify the issue.

* Can you turn on the debug logging so we can see what HTTP response is coming back from the broker?
* What is your verification configuration?
* Do you have the 'pacts for verification' API turned on in the broker? (if you don't know what this means, the answer is probably no).

**Comment by bethesque** [comment]
Actually, just to confirm, if you go to your broker, and then click on the "API browser button", can you tell me if you can find the text "pb:provider-pacts-for-verification" or "beta:provider-pacts-for-verification" in the list of relations on the left hand side of the page.

<img width="679" alt="Screen Shot 2020-10-16 at 10 26 01 am" src="https://user-images.githubusercontent.com/446228/96195999-02b92380-0f9a-11eb-8e5d-d35d956568f6.png">

**Comment by anddreiu** [comment]
Hi @bethesque. No, I can't find those 2 entries.

**Comment by uglyog** [comment]
It's the `tag=null` that is causing the NPE

**Comment by uglyog** [comment]
4.1.9 released with the fix

**Comment by anddreiu** [comment]
I can confirm that the issue is fixed in 4.1.9 version. Thank you.
